### ISSUE TITLE
Issues After Upgrade... Long Term Data issues. 

### ISSUE THREAD

**Issue by jmackxiii** [pi-hole/pi-hole#1810]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**
0

---
**Expected behaviour:**

Expecting the LongTermData portion to work/populate/load

**Actual behaviour:**

https://imgur.com/SBaBDtj 
This Screen Shot if from the Query Log section of the Long Term Data.. Nothing ever populates and version number for all three components are listed as N/A on all page footers..

**Steps to reproduce:**

Just ran the update last night when it was released.

**Debug token 6eu7qpdm3

**Troubleshooting undertaken, and/or other relevant information:**

So I ran the update last night.. At first FTL would not load properly.. Rebooted my device and it appears to run.. Cleared my browser cache and I can see my dashboard statistics just fine (they were -- before).
However at the page footer the version display is NA for Pihole/Web/FTL version (may be nothing).
Also any long term stats will not load (even when picking just today for data), it just spins and spins. Also when clicking on Query Log I get an error "An unknown error occured when loading the data"
I have done a repair install, I also do not have any custom skins, or any other add-ons for Pihole installed.
Debug Token: r6eu7qpdm3

pihole -v only shows
::: Pi-hole version is v3.2 (Latest version is v3.2)
::: Web-Admin version is v3.2 (Latest version is v3.2)

Tried different computers, different browsers, cleared cache/cookies/etc
Behavior is the same on all tested devices.

Running a repair/Reconfigure it steps through the process.. Get to the first dialog box (Repair/Reconfigure) 
  [✓] Root user check
  [i] Repair option selected
  [✓] Disk space check

  [✓] Update local cache of available packages

  [i] Checking apt-get for upgraded packages...E: Unmet dependencies. Try using -f.
  [✓] Checking apt-get for upgraded packages... up to date!

  [i] Installer Dependency checks...
  [✗] Checking for apt-utils (will be installed)
  [✗] Checking for dialog (will be installed)
  [✗] Checking for debconf (will be installed)
  [✗] Checking for dhcpcd5 (will be installed)
  [✗] Checking for git (will be installed)
  [✗] Checking for iproute (will be installed)
  [✗] Checking for whiptail (will be installed)

**Comment by Agromahdi123** [comment]
have you tried 
sudo service pihole-FTL stop
and then sudo pihole-FTL start?

**Comment by davidvermillion** [comment]
I'm having the same issue. Tried what you suggested Agromahdi123, but it did nothing.

**Comment by jmackxiii** [comment]
Just did the stop/start. Same thing.

**Comment by ghmajx** [comment]
I'm having the same issue, and the new 'clients over time' graph is blank too.

**Comment by jacobsalmela** [comment]
It seems your Pi-hole has not fully completed the update--your debug log is still from the previous version.  What happens if you `pihole -up` again?

**Comment by jmackxiii** [comment]
::: Checking for updates...
::: Pi-hole Core:   up to date
::: FTL:            up to date
::: Web Interface:  up to date
:::
::: Everything is up to date!

**Comment by jacobsalmela** [comment]
What if you
```
apt-get update
apt-get upgrade
```
then run repair/reconfigure

**Comment by Mcat12** [comment]
Check to make sure that the `php-sqlite` package (or whatever your distro names it) is installed. This should have been added in the update...

**Comment by Agromahdi123** [comment]
a thing i noticed (this happened to me but i did way to much to say what fixed it) is that the original update script said it could not find FTL on github to download, so i changed branch from whatever is default to dev, then back to master, then pihole -up the installer got it from the master then.

**Comment by Mcat12** [comment]
> [i] Checking apt-get for upgraded packages...E: Unmet dependencies. Try using -f.

**Comment by padawan** [comment]
Same issue here. I ran these steps:
- `sudo apt-get update && sudo apt-get upgrade`
- `pihole -up` (this step updated some components but failed with an error saying it couldn't update the packages cache) :

> pi@raspberrypi:~ $ pihole -up
> ::: Checking for updates...
> ::: Pi-hole Core:   update available
> ::: FTL:            update available
> ::: Web Interface:  update available
> :::
> ::: Updating Pi-hole core and web admin files
> :::
> ::: Checking for existing repository...
> :::    Updating repo in /etc/.pihole... done!
>  done!
> 
>         .;;,.
>         .ccccc:,.
>          :cccclll:.      ..,,
>           :ccccclll.   ;ooodc
>            'ccll:;ll .oooodc
>              .;cll.;;looo:.
>                  .. ','.
>                 .',,,,,,'.
>               .',,,,,,,,,,.
>             .',,,,,,,,,,,,....
>           ....''',,,,,,,'.......
>         .........  ....  .........
>         ..........      ..........
>         ..........      ..........
>         .........  ....  .........
>           ........,,,,,,,'......
>             ....',,,,,,,,,,,,.
>                .',,,,,,,,,'.
>                 .',,,,,,'.
>                   ..'''.
> 
> 
>   [✓] Root user check
>   [i] Performing unattended setup, no whiptail dialogs will be displayed
>   [✓] Disk space check
> 
>   [✗] Update local cache of available packages
>   Error: Unable to update package cache. Please try "apt-get update"Unable to complete update, contact Pi-hole

- `sudo pihole -up` (this one completed just fine with updating the remaining components that failed because of the above error)

> pi@raspberrypi:~ $ sudo pihole -up
> ::: Checking for updates...
> ::: Pi-hole Core:   up to date
> ::: FTL:            update available
> :::
> ::: FTL out of date
> 
>   [i] FTL Checks...
>   [✓] Detected ARM-hf architecture (armv6 or lower) Using ARM binary
>   [i] Checking for existing FTL binary...
>   [✓] Downloading and Installing FTL
> :::
> ::: Web Interface:  update available
> :::
> ::: Pi-hole Web Admin files out of date
>   [✓] Check for existing repository in /var/www/html/admin
>   [✓] Update repo in /var/www/html/admin
> 
> :::
> ::: Web Admin version is now at v3.2}
> ::: If you had made any changes in '/var/www/html/admin/', they have been stashed using 'git stash'
> :::
> ::: FTL version is now at vDev-437af07}
> 
>   [✓] Starting pihole-FTL service
> 
>   [✓] Enabling pihole-FTL service to start on reboot

- then ran into the issues described here.

Clearing my browser cache solved a few graphs/figures but SW versions are still showing up as N/A and longterm graphs don't show (and don't generate CPU load either).
Long Term Data > Query Log results in an alert "An unknown error occured while loading the data." 

Trying to `sudo apt-get install php-sqlite` gave an error: "E: Unable to locate package php-sqlite". you probably meant `php5-sqlite` (for PHP5).
After installing `php5-sqlite`, problems with graphs/data not loading disappeared.
You might want to look at the proper installation of `php5-sqlite` in the upgrade, that might be it.

**Comment by jmackxiii** [comment]
Output of update
pi@raspberrypi:~ $ sudo apt-get update
Hit http://mirrordirector.raspbian.org jessie InRelease
Hit http://archive.raspberrypi.org jessie InRelease
Hit http://mirrordirector.raspbian.org jessie/main armhf Packages
Hit http://archive.raspberrypi.org jessie/main armhf Packages
Hit http://archive.raspberrypi.org jessie/ui armhf Packages
Hit http://mirrordirector.raspbian.org jessie/contrib armhf Packages
Hit http://mirrordirector.raspbian.org jessie/non-free armhf Packages
Hit http://mirrordirector.raspbian.org jessie/rpi armhf Packages
Ign http://archive.raspberrypi.org jessie/main Translation-en_GB
Ign http://archive.raspberrypi.org jessie/main Translation-en
Ign http://archive.raspberrypi.org jessie/ui Translation-en_GB
Ign http://archive.raspberrypi.org jessie/ui Translation-en
Ign http://mirrordirector.raspbian.org jessie/contrib Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/contrib Translation-en
Ign http://mirrordirector.raspbian.org jessie/main Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/main Translation-en
Ign http://mirrordirector.raspbian.org jessie/non-free Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/non-free Translation-en
Ign http://mirrordirector.raspbian.org jessie/rpi Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/rpi Translation-en
Reading package lists... Done

**Comment by jmackxiii** [comment]
Upgrade
Reading package lists... Done
Building dependency tree       
Reading state information... Done
You might want to run 'apt-get -f install' to correct these.
The following packages have unmet dependencies:
 libc6 : Depends: libgcc1 but it is not installed
E: Unmet dependencies. Try using -f.

**Comment by jmackxiii** [comment]
-f install
pi@raspberrypi:~ $ sudo apt-get -f install
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following extra packages will be installed:
  libgcc1
The following NEW packages will be installed:
  libgcc1
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
2 not fully installed or removed.
Need to get 0 B/39.4 kB of archives.
After this operation, 160 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
E: Cannot get debconf version. Is debconf installed?debconf: apt-extracttemplates failed: No such file or directory
dpkg: dependency problems prevent configuration of libc6:armhf:
 libc6:armhf depends on libgcc1; however:
  Package libgcc1 is not installed.

dpkg: error processing package libc6:armhf (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of multiarch-support:
 multiarch-support depends on libc6 (>= 2.13-5); however:
  Package libc6:armhf is not configured yet.

dpkg: error processing package multiarch-support (--configure):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 libc6:armhf
 multiarch-support
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by jmackxiii** [comment]
Those are the output of what I get doing update and upgrade. Not really sure how to repair what's missing. Not that savy with Linux.

**Comment by jacobsalmela** [comment]
Yeah, you're missing some dependencies; for sure these two: `libgcc1`, `debconf`, likely more.  Pi-hole needs to be able to install everything it requires so you'll need to fix those dependency issues before the install can successfully complete.

**Comment by jmackxiii** [comment]
How can I correct those msising items?

**Comment by jacobsalmela** [comment]
Try installing those two packages first, then see if any others are listed.

**Comment by jmackxiii** [comment]
Thank you... Sorry for asking a stupid/noob question. How would I install just those items?

**Comment by jmackxiii** [comment]
Trying to install libgcc1 and get this
pi@raspberrypi:~ $ sudo apt-get install libgcc1                                 Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
 libgcc1
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
2 not fully installed or removed.
Need to get 0 B/39.4 kB of archives.
After this operation, 160 kB of additional disk space will be used.
E: Cannot get debconf version. Is debconf installed?
debconf: apt-extracttemplates failed: No such file or directory
dpkg: dependency problems prevent configuration of libc6:armhf:
libc6:armhf depends on libgcc1; however:
 Package libgcc1 is not installed.

dpkg: error processing package libc6:armhf (--configure):
dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of multiarch-support:
multiarch-support depends on libc6 (>= 2.13-5); however:
 Package libc6:armhf is not configured yet.

dpkg: error processing package multiarch-support (--configure):
dependency problems - leaving unconfigured
Errors were encountered while processing:
libc6:armhf
multiarch-support
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by Mcat12** [comment]
Are you trying to install a package from another architecture, or have you done that in the past?

**Comment by jmackxiii** [comment]
no.. at least not to my knowledge/skills.
As of right now SSH is not responding, and the PiHole web says lost connection to API.. I cannot reboot the pi (web option does nothing). I think at this point I will just reimage the device.. I have a backup from the last time I imaged this with my base configuration. Ill load that, then try to run the pihole update again.. No real other options right now.

The strange thing is it's still working (generates a tail log), blocking ads... etc... so who knows.

**Comment by Mcat12** [comment]
It seems like something messed up your packages, so restarting from a working/fresh image would help see if it's reproducible

**Comment by jmackxiii** [comment]
I will do that tonight, and report back if I still have issues.

**Comment by Agromahdi123** [comment]
once you have your image up, do a sudo apt-get update && sudo apt-get dist-upgrade -y and let it finish. the debian package manager was throwing errors because something never fully installed.  If you APT tells you "2 not fully installed or removed." first think you should do is run sudo apt-get install --fix-missing and that will finish installing packages that may have been interrupted.

**Comment by jmackxiii** [comment]
Got it, will report back with any issues.. Thanks again.

**Comment by jmackxiii** [comment]
So I have my image loaded... before I update pihole.. I am running sudo apt-get update.. and get the follow 
Get:1 http://mirrordirector.raspbian.org jessie InRelease [14.9 kB]
Get:2 http://archive.raspberrypi.org jessie InRelease [22.9 kB]
Get:3 http://mirrordirector.raspbian.org jessie/main armhf Packages [9,536 kB]
Get:4 http://archive.raspberrypi.org jessie/main armhf Packages [171 kB]
Get:5 http://archive.raspberrypi.org jessie/ui armhf Packages [58.9 kB]
Ign http://archive.raspberrypi.org jessie/main Translation-en_GB
Ign http://archive.raspberrypi.org jessie/main Translation-en
Get:6 http://mirrordirector.raspbian.org jessie/contrib armhf Packages [43.3 kB]
Ign http://archive.raspberrypi.org jessie/ui Translation-en_GB
Get:7 http://mirrordirector.raspbian.org jessie/non-free armhf Packages [84.2 kB]
Ign http://archive.raspberrypi.org jessie/ui Translation-en
Get:8 http://mirrordirector.raspbian.org jessie/rpi armhf Packages [1,356 B]
Ign http://mirrordirector.raspbian.org jessie/contrib Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/contrib Translation-en
Ign http://mirrordirector.raspbian.org jessie/main Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/main Translation-en
Ign http://mirrordirector.raspbian.org jessie/non-free Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/non-free Translation-en
Ign http://mirrordirector.raspbian.org jessie/rpi Translation-en_GB
Ign http://mirrordirector.raspbian.org jessie/rpi Translation-en
Fetched 9,932 kB in 16s (607 kB/s)
Reading package lists... Error!
E: Problem parsing dependency Depends
E: Error occurred while processing liblangtag1 (NewVersion2)
E: Problem with MergeList /var/lib/dpkg/status
E: The package lists or status file could not be parsed or opened.

**Comment by Agromahdi123** [comment]
your package manager seems broken, i dont know how to fix those particular errors, i would throw them in google search, you need to fix this before you can get pi-hole to work, its going to use apt to install what it needs, and it looks like apt cant update. Can you just start from a new fresh build of raspbian? or do you have your conf files and everything set the way you want on there?

**Comment by Agromahdi123** [comment]
sudo rm -vf /var/lib/apt/lists/*
delete old package lists
sudo apt-get update
see if that works

**Comment by jmackxiii** [comment]
The stuff here seemed to fix it

**Comment by jmackxiii** [comment]
Ok... pihole updated... Long Term Data still doesn't load...
 Your debug token is: oookdzrj6h
New Token
This looks to be a new process so the logs should be current

**Comment by jmackxiii** [comment]
When loading the tab.. I get a message.. an unknown error has occurred.. Error connecting to the database

**Comment by Agromahdi123** [comment]
sudo apt-get install php5-sqlite
pihole -r

**Comment by jmackxiii** [comment]
pi@raspberrypi:~ $ sudo apt-get install php5-sqlite
Reading package lists... Done
Building dependency tree       
Reading state information... Done
php5-sqlite is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 4 not upgraded.

And repair
[✓] Root user check
  [i] Repair option selected
  [✓] Disk space check

  [✓] Update local cache of available packages

  [✓] Checking apt-get for upgraded packages... up to date!

  [i] Installer Dependency checks...
  [✓] Checking for apt-utils
  [✓] Checking for dialog
  [✓] Checking for debconf
  [✓] Checking for dhcpcd5
  [✓] Checking for git
  [✓] Checking for iproute2
  [✓] Checking for whiptail

  [i] Performing reconfiguration, skipping download of local repos
  [✓] Resetting repository within /etc/.pihole...
  [i] Resetting repository within /var/www/html/admi  [✓] Resetting repository within /var/www/html/admin...
  [i] Main Dependency checks...
  [✓] Checking for bc
  [✓] Checking for cron
  [✓] Checking for curl
  [✓] Checking for dnsmasq
  [✓] Checking for dnsutils
  [✓] Checking for iputils-ping
  [✓] Checking for lsof
  [✓] Checking for netcat
  [✓] Checking for sudo
  [✓] Checking for unzip
  [✓] Checking for wget
  [✓] Checking for idn2
  [✓] Checking for lighttpd
  [✓] Checking for php5-common
  [✓] Checking for php5-cgi
  [✓] Checking for php5-sqlite

  [✓] Installing scripts from /etc/.pihole

  [i] Installing configs from /etc/.pihole...
  [i] Existing dnsmasq.conf found... it is not a Pi-hole file, leaving alone!
  [i] Copying 01-pihole.conf to /etc/dnsmasq.d/01-pi  [✓] Copying 01-pihole.conf to /etc/dnsmasq.d/01-pihole.conf

  [i] Creating log and changing owner to dnsmasq... log already exists!

  [i] Installing blocking page...
  [i] Creating directory for blocking page, and copy  [✓] Creating directory for blocking page, and copying files
  [✗] Backing up index.lighttpd.html
      No default index.lighttpd.html file found... not backing up

  [✓] Installing sudoer file

  [✓] Installing latest Cron script

  [✓] Installing latest logrotate script

  [i] FTL Checks...
  [✓] Detected ARM-hf architecture (armv7+)
  [i] Checking for existing FTL binary...
  [i] Latest FTL Binary already installed (v2.12). Confirming Checksum...
  [i] Checksum correct. No need to download!
  [i] Restarting services...

  [✓] Starting dnsmasq service

  [✓] Enabling dnsmasq service to start on reboot

  [✓] Starting lighttpd service

  [i] Enabling lighttpd service to start on reboot..  [✓] Enabling lighttpd service to start on reboot

  [✓] Starting pihole-FTL service

  [i] Enabling pihole-FTL service to start on reboot  [✓] Enabling pihole-FTL service to start on reboot
  [i] Preparing to run gravity.sh to refresh hosts...
  [i] Cleaning up previous install (preserving whitelist/blacklist)
  [i] Running gravity.sh
  [i] Neutrino emissions detected...
  [✓] Pulling blocklist source list into range

  [i] Target: raw.githubusercontent.com (hosts)
  [✓] Status: Retrieval successful

  [i] Target: mirror1.malwaredomains.com (justdomains)
  [✓] Status: Retrieval successful

  [i] Target: sysctl.org (hosts)
  [✓] Status: Retrieval successful

  [i] Target: zeustracker.abuse.ch (blocklist.php?download=domainblocklist)
  [✓] Status: Retrieval successful

  [i] Target: s3.amazonaws.com (simple_tracking.txt)  [✓] Status: Retrieval successful

  [i] Target: s3.amazonaws.com (simple_ad.txt)
  [✓] Status: Retrieval successful

  [i] Target: hosts-file.net (ad_servers.txt)
  [✓] Status: Retrieval successful

  [✓] Consolidating blocklists
  [✓] Extracting domains from blocklists
  [i] 129,154 domains being pulled in by gravity
  [✓] Removing duplicate domains
  [i] 106,012 unique domains trapped in the Event Horizon

  [i] Adding 6 blocklist source domains to the white  [✓] Adding 6 blocklist source domains to the whitelist
  [✓] Whitelisting 7 domains
  [✓] Parsing domains into hosts format
  [✓] Cleaning up stray matter

  [✓] Force-reloading DNS service
  [✓] DNS service is running
  [✓] Pi-hole blocking is Enabled

  [i] The install log is located at: /etc/pihole/install.log
  Update Complete! 
pi@raspberrypi:~ $

**Comment by jacobsalmela** [comment]
Can you flush your browser cache just to rule that out?

**Comment by davidvermillion** [comment]
Yes, after I did a total reinstall, I had to do a force refresh and the problem went away.

**Comment by jmackxiii** [comment]
Tried clearing cache, tried multiple computers and browsers... same thing..

In LTD the Graphics and Top Query Portion just loads nothing (Just spins, or returns nothing), even when explicitly selecting "Today" .. 

Query Log gives the error "An Unknown error occured while loading the data. Error Connecting to database.

**Comment by jacobsalmela** [comment]
OK generate a new debug log @jmackxiii and we can take a look

**Comment by jmackxiii** [comment]
Just made a new set
Your debug token is: fk3eu3au1a

**Comment by jacobsalmela** [comment]
Everything in your new log looks good.  

Try this:
```
systemctl restart pihole-FTL
```

and then will you power cycle the entire server to rule that out the command above doesn't work?

**Comment by jmackxiii** [comment]
I tried to run that command.
Failed to restart pihole-FTL.service: Access denied
I had to run it as "sudo systemctl restart pihole-FTL"

Rebooting the device...

For what it's worth in the dashboard FTL stated it was running.

**Comment by jmackxiii** [comment]
Same behavior after a reboot also.

**Comment by jacobsalmela** [comment]
```
pihole uninstall
curl -sSL https://install.pi-hole.net | bash
```
?

**Comment by jmackxiii** [comment]
Running now.

**Comment by jmackxiii** [comment]
Success! Thank you for your help!

**Comment by jmackxiii** [comment]
One thing I did notice, and it's not a big deal..

But I configured it for both IPv4 and IPv6... however in the Network Information only IPv4 is listed. (even did a repair/reconfigure to make sure it was selected)

**Comment by jacobsalmela** [comment]
Great; glad it's working.  You can open a new issue if you'd like.
